README.md

```diff
@@ -489,6 +489,32 @@ docker run -d \
     simplelogin/app:3.4.0 python job_runner.py
 ```
 
+#### Enable Sending via SMTP
+
+For sending via SMTP  you need to make sure you have the required SSL certificate and key at following, as SMTP requires SSL.
+
+- `$(pwd)/smtp_ssl_cert.pem` for SSL Certificate file,
+- `$(pwd)/smtp_ssl_privkey.key` for Key file.
+- adjust this according to where you have the files present.
+
+then run `SMTP handler`
+
+```bash
+docker run -d \
+    --name sl-smtp \
+    -v $(pwd)/sl:/sl \
+    -v $(pwd)/sl/upload:/code/static/upload \
+    -v $(pwd)/simplelogin.env:/code/.env \
+    -v $(pwd)/dkim.key:/dkim.key \
+    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
+    -v $(pwd)/smtp_ssl_cert.pem:/smtp_ssl_cert.pem
+    -v $(pwd)/smtp_ssl_privkey.key:/smtp_ssl_privkey.key
+    -p 465:465 \
```

Comment:
Not sure how do it via Postfix.
Just out of curiosity, how would protecting this port via postfix prevent attacks ?

Reply:
Postfix can block requests from bots quite efficiently. And if we add spam scanners like SpamAssassin or rspamd to Postfix, we can be quite sure that requests that reach the SMTP handler is legit.

Reply:
Oh ok. Gotcha. 
But the authentication should be the first thing that happens, so does Postfix allows for a programmatic way of handling authentication ? I couldn't seem to find any good way to handle auth in postfix's docs.

One more way to handle bots is that we could use Fail2ban for this post, they allow for creation of custom filters quite easily. Then later we are sending it to postfix anyways, there it can handle SPAM.

So Basically the flow can be like:
Port 465 (Protected by Fail2Ban) -> SMTP_Handler.py -> Postfix (SPAM handling) -> Final Recipient

Reply:
> But the authentication should be the first thing that happens, so does Postfix allows for a programmatic way of handling authentication ? I couldn't seem to find any good way to handle auth in postfix's docs.

That's a good question and we're actually looking for the answer as well :).

Reply:
I'm not familiar with fail2ban unfortunately. I think we still need Postfix to handle TLS, grey listing, spam scanning, slow connections that take up workers and all other edge cases due to the email protocol. One way to have Postfix is:
1) postfix listening on port 587/465 that requires SMTP AUTH
2) postfix talks to Python to verify whether the authentication is correct. This can be tricky as as postfix only support sasl authentication using dovecot and cyrus.
3) If authentication is correct, Postfix relays to SMTP handler

It'd be great if we can modify the PR to use this pattern.

Reply:
> I'm not familiar with fail2ban unfortunately. 

Fail2ban basically detects the authentication failures and bans the corresponding IP. the ban time can be set. This is very useful to prevent brute-force attacks. You can check that out here: https://www.fail2ban.org/


> 2. postfix talks to Python to verify whether the authentication is correct. This can be tricky as as postfix only support sasl authentication using dovecot and cyrus.

Yes, I as well, couldn't figure out how to make postfix talk to python. 


> It'd be great if we can modify the PR to use this pattern.

That would certainly be great. We would somehow need to figure out the "2." part.